### DISCUSSION TITLE
Return code for AP's channel

### DISCUSSION THREAD

**Discussion by treno234** [aircrack-ng/aircrack-ng#2358] [Q&A]
Hi,

I am trying to write a script that would change the channel of my wifi when it detects the enemy changed broadcasting channel. I am not connected to this wifi. It seems like this functionality has already been built into aireplay-ng (_"wlan0 is on channel X, but the AP uses channel Y"_) but I don't know how to use it. Can aireplay return this channel info as a return value? That way you could simply check every so often if the channel is still the same and adjust accordingly.

**Comment by Mister-X-** [comment]
What do you mean by `change the channel of my wifi`? What are you referring to? An AP, a wireless card in monitor mode? I'll go with the assumption you want that in airodump-ng.

No, aireplay-ng doesn't return this channel as a return value.

You'll have to elaborate a bit more.

**Reply by treno234** [reply]
Yes. In order for aireplay-ng to work properly the card needs to be in monitor mode and on the same channel as the AP I am injecting packets to. Otherwise you get that _"wlan0 is on channel X, but the AP uses channel Y"_ message. 
Right now I use the script you can find on the internet where aireplay-ng sends 100 deauth packets and then sleeps for certain amount of time within a loop. I want to change it so it retrieves the AP's channel at the start of the loop (so basically after the sleep command) so I can change my wifi channel to the AP's with the `iwconfig wlan0 mode monitor channel Y` command.

How do I get the AP's channel as a usable return value? I am not linux savy but have done some basic c++ programming in the past. It's a real shame aireplay-ng doesn't do that since it obviously knows the new channel location of the AP you are targeting.

**Reply by Mister-X-** [reply]
Return codes are meant to indicate success or failure.

Aireplay-ng sometimes know, it's not foolproof. It's likely because the channel is close enough in terms of frequencies, and the frame from an adjacent channel (plus or minus one or two)

Since you're familiar with C++, then you could use [libtins](https://libtins.github.io) to do it.
